plugins/etcd/src/main/java/org/radargun/service/EtcdService.java

```diff
@@ -17,17 +25,59 @@
  * Note that etcd saves some data in the distribution directory, and you need to delete it between runs. Using the distributionZip
  * property is preferred, so the distribution is 'clean' with each run.
  *
+ * The documentation about the the configuration is available at https://coreos.com/etcd/docs/latest/op-guide/configuration.html
+ *
  */
 @Service(doc = "Etcd")
 public class EtcdService implements Lifecycle {
 
    private Process server;
 
-   @Property(doc = "Directory of the etcd distribution.", optional = false)
+   @Property(doc = "Distribution zip. If set, the zip will be extracted to the 'distributionDir' directory.", optional = false)
+   protected String distributionZip;
+
+   /**
+    * It is realted with the --discovery member flag.
+    */
+   @Property(doc = "Prefix flags need to be set when using discovery service.", optional = false)
+   protected String discovery;
+
+   @Property(doc = "Server ip address.", optional = false)
+   protected String ip;
+
+   @Property(doc = "Directory of the etcd distribution.")
    protected String distributionDir;
 
-   @Property(doc = "Distribution zip. If set, the zip will be extracted to the 'distributionDir' directory.")
-   protected String distributionZip;
+   /**
+    * It is related with the --listen-port-urls member flag. In this case the hostname will be dynamically full filled.
+    */
+   @Property(doc = "The listen port number. Default is 2379")
+   protected int listenClientPort = 2379;
+
+   /**
+    * It is related with the --listen-peer-urls member flag. In this case the hostname will be dynamically full filled.
+    */
+   @Property(doc = "The listen peer port port number. Default is 2380")
+   protected int listenPeerPort = 2380;
+
+   /**
+    * It is realted with the --name member flag.
+    */
+   @Property(doc = "The server name. Each server must have an unique name")
+   protected String name;
```

Comment:
Is name really optional? You're later passing it to the command and it could be null at that point.